packages/kbn-optimizer/src/optimizer/kibana_platform_plugins.ts

```diff
@@ -36,7 +36,7 @@ export function findKibanaPlatformPlugins(scanDirs: string[], paths: string[]) {
     .sync(
       Array.from(
         new Set([
-          ...scanDirs.map((dir) => `${dir}/*/kibana.json`),
+          ...scanDirs.map((dir) => `${dir}/**/kibana.json`),
```

Comment:
Wonder if we could have some sort of maxDepth by using a (very unreadable) glob like:

```
`${dir}/{*,*/*,*/*/*,*/*/*/*,*/*/*/*/*}/kibana.json`
```

Reply:
Either that or 

```ts
 ...scanDirs
            .map((dir) => [
              `${dir}/*/kibana.json`,
              `${dir}/*/*/kibana.json`,
              `${dir}/*/*/*/kibana.json`,
              `${dir}/*/*/*/*/kibana.json`,
              `${dir}/*/*/*/*/*/kibana.json`,
            ])
            .flat(),
```

Which is a little longer (even if it can be extracted to a function), but probably more explicit.

Not 100% sure this is really needed (as it doesn't address the second point), but I have no problem implementing that if we think it's a little better can current implem.

wdyt?

Reply:
Let's err on the side of caution here and go ahead and do that. Optimizer is already slow as it is, let's be defensive.